### DISCUSSION TITLE
Vector `to_array` and `as_*` functions by ref vs value

### DISCUSSION THREAD

**Discussion by Ababwa** [bitshifter/glam-rs#567] [General]
Is there a reason the `to_array` and `as_*` vector functions are by reference? All other vector functions that do not mutate use values.
My case is probably niche, but I happen to be using unaligned vectors, and cannot call by-reference functions on them.
I assume it would be a breaking change to change those functions to use values at this point, but I am curious about the rationale for the original decision.
Thank you

**Comment by bitshifter** [comment]
Sorry I missed this or forgot to respond. It was probably 6 years ago that I added these and I don't know if there was a rational.
Hmm, it is inconsistent with most of the rest of the API. 

It would be a breaking change but it might not break much actual code in the real world.

**Comment by bitshifter** [comment]
Created an issue to do a consistency pass on this.
